### PR TITLE
Fix ``_ErrorLog`` not iterable

### PR DESCRIPTION
**Pull request by nsoranzo**
Fix the following mypy error:
```
lib/galaxy/tool_util/linters/xsd.py:28: error: "_ErrorLog" has no attribute
"__iter__" (not iterable)  [attr-defined]
                for error in xsd.error_log:
                             ^~~~~~~~~~~~~
```

xref: https://github.com/galaxyproject/galaxy/pull/17081/files#r1490662640

### CODE REVIEW

File: lxml-stubs/etree.pyi [hunk 1]
```diff
@@ -577,7 +577,12 @@ def tostring(
     inclusive_ns_prefixes: Any = ...,
 ) -> _AnyStr: ...
 
-class _ErrorLog: ...
+class _LogEntry:
+    path: Optional[str]
```
Comment (scoder) [inline lxml-stubs/etree.pyi:581]: Adding one attribute will lead to errors about all other attributes. Could you please add the complete list of attributes? There are also several properties.
https://github.com/lxml/lxml/blob/97e6a1a0b4fa39662fd6fd4b49a348619cde165a/src/lxml/xmlerror.pxi#L52-L56

File: lxml-stubs/etree.pyi [hunk 2]
```diff
@@ -577,7 +577,12 @@ def tostring(
...
+
+class _ErrorLog:
+    def __iter__(self) -> Iterator["_LogEntry"]: ...
```
Comment (scoder) [inline lxml-stubs/etree.pyi:584]: It's actually not `_ErrorLog` that's iterable but its base class. Also, again, there are lots of public attributes and methods missing here. It's best to add all of them.

Comment (nsoranzo) [reply lxml-stubs/etree.pyi]: Thanks, I think I've added all the public attributes/properties/methods listed at https://lxml.de/apidoc/lxml.etree.html for `_LogEntry`, `_ErrorLog` and its parent classes. Let me know if I've missed some.

**Comment by scoder** [comment]
Thanks

**Pull request merged**